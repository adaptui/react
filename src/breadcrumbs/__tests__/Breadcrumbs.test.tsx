import * as React from "react";
import { axe, render } from "reakit-test-utils";

import { Breadcrumbs, BreadcrumbLink } from "..";

const BreadcrumbComp = () => {
  return (
    <Breadcrumbs className="breadcrumb">
      <ol>
        <li>
          <BreadcrumbLink href="https://www.w3.org/TR/wai-aria-practices-1.1/">
            WAI-ARIA Authoring Practices 1.1
          </BreadcrumbLink>
        </li>
        <li>
          <BreadcrumbLink as="span" onClick={() => alert("Go to link")}>
            Design Patterns
          </BreadcrumbLink>
        </li>
        <li>
          <BreadcrumbLink
            isCurrent
            href="https://www.w3.org/TR/wai-aria-practices-1.1/#breadcrumb"
          >
            Breadcrumb Pattern
          </BreadcrumbLink>
        </li>
        <li>
          <BreadcrumbLink href="https://www.w3.org/TR/wai-aria-practices-1.1/examples/breadcrumb/index.html">
            Breadcrumb Example
          </BreadcrumbLink>
        </li>
      </ol>
    </Breadcrumbs>
  );
};

describe("Breadcrumb", () => {
  it("should render correctly", () => {
    const { getByTestId: testId, baseElement } = render(<BreadcrumbComp />);

    expect(baseElement).toMatchInlineSnapshot(`
      <body>
        <div>
          <nav
            aria-label="Breadcrumb"
            class="breadcrumb"
          >
            <ol>
              <li>
                <a
                  href="https://www.w3.org/TR/wai-aria-practices-1.1/"
                >
                  WAI-ARIA Authoring Practices 1.1
                </a>
              </li>
              <li>
                <span
                  role="link"
                  tabindex="0"
                >
                  Design Patterns
                </span>
              </li>
              <li>
                <a
                  aria-current="page"
                  href="https://www.w3.org/TR/wai-aria-practices-1.1/#breadcrumb"
                >
                  Breadcrumb Pattern
                </a>
              </li>
              <li>
                <a
                  href="https://www.w3.org/TR/wai-aria-practices-1.1/examples/breadcrumb/index.html"
                >
                  Breadcrumb Example
                </a>
              </li>
            </ol>
          </nav>
        </div>
      </body>
    `);
  });

  test("Breadcrumb renders with no a11y violations", async () => {
    const { container } = render(<BreadcrumbComp />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
