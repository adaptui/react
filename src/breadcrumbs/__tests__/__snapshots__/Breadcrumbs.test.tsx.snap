// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render correctly 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumb"
  >
    <ol>
      <li>
        <a
          href="https://www.w3.org/TR/wai-aria-practices-1.1/"
        >
          WAI-ARIA Authoring Practices 1.1
        </a>
      </li>
      <li>
        <a
          href="https://www.w3.org/TR/wai-aria-practices-1.1/#aria_ex"
        >
          Design Patterns
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          href="https://www.w3.org/TR/wai-aria-practices-1.1/#breadcrumb"
        >
          Breadcrumb Pattern
        </a>
      </li>
      <li>
        <a
          href="https://www.w3.org/TR/wai-aria-practices-1.1/examples/breadcrumb/index.html"
        >
          Breadcrumb Example
        </a>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
